src/components/FunctionSeries.tsx

```diff
@@ -0,0 +1,33 @@
+import { useMemo } from 'react';
+
+import { usePlotContext } from '../contexts/plotContext';
+import { SeriesPoint } from '../types';
+import { toNumber } from '../utils';
+
+import { LineSeries, LineSeriesProps } from './LineSeries';
+
+export interface FunctionSeriesProps extends Omit<LineSeriesProps, 'data'> {
+  getY: (x: number) => number;
+  step?: number;
+}
+
+export function FunctionSeries(props: FunctionSeriesProps) {
+  const { getY, step = 0.1, ...otherProps } = props;
```

Comment:
@lpatiny do you think it makes sense to have a `step` option ? Anyway I think it should default to `1`, because unless the user zooms inside the page, 1=1 pixel.

Reply:
I'm ok with calculating it, but how? What value other than 1 point per unit (meaning a step of 1) do you expect?

Reply:
We can compute a better value using `Math.ceil(globalThis.devicePixelRatio || 1)` to take page zoom into account.

Reply:
step=1 result 
![image](https://user-images.githubusercontent.com/56769876/151829213-43e7b3a0-84ff-440a-98de-b66e150c3a28.png)

Reply:
Ok, then I misunderstood what `step` means. What I think it should be:
if the plotWidth is `850` for example, we should have `850 * devicePixelRatio` computed points.

Reply:
Basically, you need to invert each x using the scale, not only the first one. Otherwise it won't work properly on non-linear scales.

Reply:
i think 1pixel is not visible anyways so we can fix step to 1
when we zoom it still 1 px